modules/LB/LB.tf

```diff
@@ -63,9 +73,9 @@ resource "aws_lb_target_group" "TG" {
 }
 
 resource "aws_route53_record" "my_dns_record" {
-  zone_id = "Z05599622K22XV64ERVX6"
-  name    = "mentorship"
+  zone_id = data.aws_route53_zone.hosted_zone.id
+  name    = "mentorship.itsguts.com"
   type    = "CNAME"
   ttl     = 300
-  records = ["my-alb1-713027817.us-east-1.elb.amazonaws.com"]
+  records = [data.aws_lb.my_alb.id]
```

Comment:
your record configuration is wrong. could you please change `records = [data.aws_lb.my_alb.id]` to` records = [aws_lb.my_alb.dns_name]`